core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/CloneWorkspaceFileTransferMonitor.scala

```diff
@@ -66,6 +66,24 @@ class CloneWorkspaceFileTransferMonitorActor(val dataSource: SlickDataSource,
         .unsafeToFuture()
     } yield ()
 
+  private def attemptTransfer(
+    pendingTransfer: PendingCloneWorkspaceFileTransfer
+  )(implicit executionContext: ExecutionContext): Future[List[Option[StorageObject]]] = {
+    val transferExpired = pendingTransfer.created.isBefore(DateTime.now().minusDays(1))
+    for {
+      copiedObjects <-
+        if (!transferExpired) copyBucketFiles(pendingTransfer)
+        else {
+          logger.warn(
+            s"File transfer from ${pendingTransfer.sourceWorkspaceBucketName} to ${pendingTransfer.destWorkspaceBucketName} did not succeed within allowed time and will no longer be retried. [workspaceId=${pendingTransfer.destWorkspaceId}]"
+          )
+          Future.successful(List.empty)
+        }
+
+      _ <- markTransferAsComplete(pendingTransfer, transferSucceeded = !transferExpired)
```

Comment:
This addresses cases where a source workspace's files can't be copied to a destination workspace for longer than a day due to persistent errors. When that happens, the monitor currently tries to clone the files nonstop and fills up our logs unnecessarily.

It doesn't attempt to protect against long running copy operations while transferring large buckets. While it would be nice to have those protections, the solution is likely to switch to STS which would be more involved.

Reply:
Oh, so `copyBucketFIles` would throw an exception at some point, and then the monitor kicks off this code again?

Reply:
Yes, that's correct. And if `copyBucketFiles` continues to throw exceptions for over a day for a given workspace, this will make it so the monitor stops trying to transfer the files for that workspace

Reply:
I think the idea is to stop _retries_, not to interrupt an in-progress operation?
And then if it fails, this entire function will return a failed future (it won't even get to `markTransferAsComplete`).